python/ray/util/spark/cluster_init.py

```diff
@@ -1226,9 +1301,24 @@ def setup_ray_cluster(
         # started cluster.
         _active_ray_cluster = cluster
 
-    head_ip = cluster.address.split(":")[0]
-    remote_connection_address = f"ray://{head_ip}:{cluster.ray_client_server_port}"
-    return cluster.address, remote_connection_address
+    if is_global:
+        # If in global mode, `setup_ray_cluster` API is blocking, once the API call
```

Comment:
Actually why do we make it blocking?

Reply:
We want to enforce that only the "caller" can terminate the global cluster (see https://github.com/ray-project/ray/pull/41153#discussion_r1394424737) . So blocking behavior makes the semantic apparent to user. :)

Reply:
or I can add a separate API like:
```
serve_ray_global_cluster(...)
```
to be blocking mode, internally it calls `setup_ray_cluster(..., is_global=True)`

this might be more clear.

Reply:
Even if you are blocking, you cannot prevent users from calling `shutdown_ray_cluster()`?

Reply:
> Even if you are blocking, you cannot prevent users from calling shutdown_ray_cluster()?

If the user is in another python process, then he cannot call valid `shutdown_ray_cluster` because there is no active ray cluster variable in his python process.

If the user is in the same python process, then the user is legitimate to shut down the Ray cluster.

For databricks user, we have another reason:

Running a blocking command, it will force the notebook to be active status unless user explicitly cancel it. The active status remind user that there is a global Ray cluster running on this notebook and when the user cancel the notebook command it means he shut down the global cluster. This provide better user experience, otherwise user might create a global Ray cluster and then forget to shut down it, then the cluster resources occupied by the global Ray cluster cannot be released.

Reply:
sgtm